In [9]:
import pandas as pd
import numpy as np

df = pd.read_csv('EV_files/EV_train.csv')
df.set_index('House ID', inplace=True)

houses_w_nans = df[pd.isnull(df).any(axis=1)]
houses_missing_data = houses_w_nans.index.tolist()

df = df[~df.index.isin(houses_missing_data)]

In [10]:
df.head(3)

Interval_1  Interval_2  Interval_3  Interval_4  Interval_5  \
House ID                                                               
11655099       0.950       0.826       0.361       0.238       0.342   
11633257       0.353       0.327       0.358       0.292       0.285   
11651552       0.150       0.181       0.150       0.150       0.131   

          Interval_6  Interval_7  Interval_8  Interval_9  Interval_10  \
House ID                                                                
11655099       0.233       0.351       0.194       0.292        0.234   
11633257       0.304       0.361       0.342       0.355        0.348   
11651552       0.125       0.088       0.106       0.094        1.019   

              ...        Interval_2871  Interval_2872  Interval_2873  \
House ID      ...                                                      
11655099      ...                0.664          0.783          0.601   
11633257      ...                0.536          0.558          0.622   
11651552      ...                2.125          0.881          0.481   

          Interval_2874  Interval_2875  Interval_2876  Interval_2877  \
House ID                                                               
11655099          0.639          0.417          0.439          0.226   
11633257          0.634          0.513          0.421          0.273   
11651552          1.194          0.138          0.119          0.038   

          Interval_2878  Interval_2879  Interval_2880  
House ID                                               
11655099          0.190          0.710          0.728  
11633257          0.296          0.291          0.289  
11651552          0.088          0.056          0.113  

[3 rows x 2880 columns]

In [11]:
df.shape

(1586, 2880)

Let's calculate the total demand for these 1586 homes and see if we can use a forecasting model to predict any weekly and half hourly periodicity in the data.

In [17]:
df.append(df.sum(), ignore_index=True)

Interval_1   Interval_2   Interval_3   Interval_4  Interval_5  \
0        0.950000     0.826000     0.361000     0.238000       0.342   
1        0.353000     0.327000     0.358000     0.292000       0.285   
2        0.150000     0.181000     0.150000     0.150000       0.131   
3        2.088000     2.075000     2.121000     2.098000       2.046   
4        1.416000     1.250000     1.270000     1.258000       1.239   
5        1.124000     0.995000     0.792000     1.256000       0.996   
6        0.103000     0.126000     0.237000     0.209000       0.183   
7        0.685000     0.691000     0.701000     0.689000       0.712   
8        1.113000     1.098000     1.355000     1.513000       1.503   
9        0.433000     0.412000     0.378000     0.393000       0.399   
10       0.771000     0.319000     0.342000     0.292000       0.341   
11       0.289000     0.709000     0.223000     0.184000       0.175   
12       0.149000     0.161000     0.096000     0.119000       0.054   
13       0.834000     0.451000     0.393000     0.410000       0.377   
14       1.037000     1.040000     1.034000     1.031000       1.100   
15       1.273000     1.213000     1.233000     1.224000       1.284   
16       0.626000     0.429000     0.219000     0.195000       0.194   
17       0.084000     0.084000     0.102000     0.128000       0.125   
18       0.661000     0.970000     0.863000     0.859000       0.125   
19       0.225000     0.225000     0.213000     0.206000       0.250   
20       0.883000     0.461000     0.783000     0.401000       0.719   
21       0.169000     0.188000     0.188000     0.200000       0.169   
22       2.061000     2.094000     2.076000     2.088000       2.119   
23       0.325000     0.281000     0.331000     0.313000       0.294   
24       0.363000     0.543000     0.362000     0.377000       0.212   
25       0.962000     0.993000     0.988000     0.415000       0.060   
26       0.188000     0.181000     0.213000     0.175000       0.106   
27       0.263000     0.368000     1.786000     2.388000       0.634   
28       4.178000     4.133000     4.850000     5.203000       4.630   
29       1.560000     1.465000     1.508000     1.550000       1.440   
...           ...          ...          ...          ...         ...   
1557     0.188000     0.125000     0.200000     0.588000       0.813   
1558     0.345000     0.361000     0.366000     0.271000       0.252   
1559     0.319000     0.319000     0.306000     0.338000       0.413   
1560     1.116000     1.121000     1.125000     1.122000       1.123   
1561     0.364000     0.401000     0.363000     0.361000       0.234   
1562     0.581000     0.814000     0.700000     0.757000       0.662   
1563     0.737000     0.753000     0.743000     0.729000       0.746   
1564     0.936000     0.878000     0.939000     0.936000       0.902   
1565     0.301000     0.299000     0.622000     0.324000       0.270   
1566     0.419000     0.772000     0.533000     0.180000       0.230   
1567     0.222000     0.214000     0.219000     0.209000       0.206   
1568     0.814000     1.039000     0.684000     0.459000       0.351   
1569     1.143000     1.120000     1.094000     1.097000       1.083   
1570     1.060000     1.060000     1.270000     0.870000       0.890   
1571     0.297000     0.296000     0.297000     0.637000       2.474   
1572     1.442000     1.280000     1.205000     1.225000       1.214   
1573     1.457000     1.708000     1.516000     1.919000       1.441   
1574     0.850000     2.276000     0.150000     0.148000       0.161   
1575     0.742000     0.527000     0.594000     0.619000       0.627   
1576     0.037000     0.026000     0.054000     0.046000       0.026   
1577     0.535000     0.529000     0.522000     0.670000       2.160   
1578     1.061000     1.018000     0.573000     0.543000       0.563   
1579     0.410000     0.285000     0.290000     0.265000       0.305   
1580     0.569000     0.574000     0.250000    

2880 30-min intervals span 60 days. Likely, not enough to capture seasonality but day of the week indicators or weekday/weekend dummy variables should certainly be applicable. Unfortunately, I don't have any dates associated with the variables to be apple to apply historical temperature data.